### PR TITLE
#169817844 Users should be able to see his/her trip statistics

### PR DESCRIPTION
**Pull request by jo-rdan**
#### What does this PR do?
- A user on barefoot nomad should be able to see the number of trips he/she created in a given time
#### Description of Task to be completed?
- Add actions to fetch data from the api 
- add reducers to handle actions and update the state
- add responsiveness to the design across many devices
#### How should this be manually tested?
- clone this repo
- checkout to this branch and make sure you have all the packages installed by typing `npm install`
- run the server locally by typing `npm start-dev` and it should open in your favorite browser
- click on the statistics link and you can change date to see the number of trips from that date
#### Any background context you want to provide?
- this PR is not yet in production
#### What are the relevant pivotal tracker stories?
[#169817844](https://www.pivotaltracker.com/story/show/169817844)
#### Screenshots (if appropriate)
image

### CODE REVIEW

File: __tests__/tripRequestTest/tripRequestAction.test.js [hunk 1]
```diff
@@ -1,108 +1,124 @@
...
+            }
+          }
+        }
+      });
+    });
+    const store = mockStore({});
+    return store
+      .dispatch(requestTrip("1", "2", "2", "2020-05-20", "0788787273"))
+      .then(() => {
+        expect(store.getActions());
```
Comment (bdushimi) [inline __tests__/tripRequestTest/tripRequestAction.test.js:52]: @jordan any reason why you're not making an assertion right here?

File: __tests__/tripRequestTest/tripRequestAction.test.js [hunk 2]
```diff
@@ -1,108 +1,124 @@
...
+              { id: "1", name: "marriot" },
+              { id: "1", name: "marriot" }
+            ]
+          }
+        }
+      });
+    });
+    const store = mockStore({});
+    return store.dispatch(GetAccomodations("1")).then(() => {
+      expect(store.getActions());
```
Comment (bdushimi) [inline __tests__/tripRequestTest/tripRequestAction.test.js:73]: @jordan any reason why you're not making an assertion right here?

File: __tests__/tripRequestTest/tripRequestAction.test.js [hunk 3]
```diff
@@ -1,108 +1,124 @@
...
+        status: 201,
+        response: {
+          status: 201,
+          data: [{ id: "1", name: "marriot" }]
+        }
+      });
+    });
+    const store = mockStore({});
+    return store.dispatch(GetAccomodations("1")).then(() => {
+      expect(store.getActions());
```
Comment (bdushimi) [inline __tests__/tripRequestTest/tripRequestAction.test.js:89]: @jordan any reason why you're not making an assertion right here?

File: __tests__/tripRequestTest/tripRequestAction.test.js [hunk 4]
```diff
@@ -1,108 +1,124 @@
...
+        status: 201,
+        response: {
+          status: 201,
+          error: "All Supported Locations"
+        }
+      });
+    });
+    const store = mockStore({});
+    return store.dispatch(GetLocations("")).then(() => {
+      expect(store.getActions());
```
Comment (bdushimi) [inline __tests__/tripRequestTest/tripRequestAction.test.js:105]: @jordan any reason why you're not making an assertion right here?

File: __tests__/tripRequestTest/tripRequestAction.test.js [hunk 5]
```diff
@@ -1,108 +1,124 @@
...
+        status: 404,
+        response: {
+          status: 404,
+          error: "Trip Unsccessfully created"
+        }
+      });
+    });
+    const store = mockStore({});
+    return store.dispatch(requestTrip("")).catch(() => {
+      expect(store.getActions());
```
Comment (bdushimi) [inline __tests__/tripRequestTest/tripRequestAction.test.js:121]: @jordan any reason why you're not making an assertion right here?

**Review by bdushimi — CHANGES_REQUESTED** [review]

File: __tests__/tripRequestTest/tripRequestAction.test.js (hunk 1 above)
Comment (jo-rdan) [reply __tests__/tripRequestTest/tripRequestAction.test.js:52]: @bdushimi , as you can see these tests are not from my feature, the reason why you see the these files in my file changes is that these tests were failing on my branch and I had to fix that. they did not have any assertions before.

**Review by bdushimi — APPROVED** [review]

**Pull request merged**